django-s3-file-field-client/src/client.ts

```diff
@@ -0,0 +1,123 @@
+import axios from 'axios';
+
+// Description of a part from initializeUpload()
+interface PartInfo {
+  part_number: number;
+  size: number;
+  upload_url: string;
+}
+// Description of the upload from initializeUpload()
+interface MultipartInfo {
+  object_key: string;
+  upload_id: string;
+  parts: PartInfo[];
+}
+// Description of a part which has been uploaded by uploadPart()
+interface UploadedPart {
+  part_number: number;
+  size: number;
+  etag: string;
+}
+// Return value from uploadFile()
+export interface UploadResult {
+  value: string;
+  state: 'aborted' | 'successful' | 'error';
+}
+
+export default class Client {
```

Comment:
I realize the name is lost because it's a default export, but maybe `S3FFClient` is a better name for the public interface?